### DISCUSSION TITLE
GitLab group bot GPG key sign

### DISCUSSION THREAD

**Discussion by weikangchia** [renovatebot/renovate#22998] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.143.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

May I know if anyone knows/is it even possible how to set up the GPG key for GitLab Bot (internal user created when we create a group access token) to be used as the Git commit user for renovate?

### What you tried so far.

Tried searching online and doesn't seems to find anything

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Does that bot have a _valid_ email address associated with it? You would need to have a valid email address plus a location where you can publish that account's public key. Without it, GPG signing won't work.

Instead, you could use an alternative email in `gitAuthor` belonging to an account which has published its public key.

**Reply by weikangchia** [reply]
I am trying to use my personal account instead and met this error. Do you happen to know what could be the issue?

`Command failed: gpg --import /tmp/git-private.key\ngpg: key XXX: \"XXX <XXX@YYY.com>\" not changed\ngpg: key XXX/XXX: error sending to agent: Inappropriate ioctl for device\ngpg: key XXX/XXX: error sending to agent: Inappropriate ioctl for device\ngpg: error reading '/tmp/git-private.key': Inappropriate ioctl for device\ngpg: import from '/tmp/git-private.key' failed: Inappropriate ioctl for device\ngpg: Total number processed: 0\ngpg:              unchanged: 1\ngpg:       secret keys read: 1\n",
         "stack": "ExecError: Command failed: gpg --import /tmp/git-private.key\ngpg: key XXX: \"XXX <XXX@YYY.com>\" not changed\ngpg: key XXX/XXX: error sending to agent: Inappropriate ioctl for device\ngpg: key XXX/XXX: error sending to agent: Inappropriate ioctl for device\ngpg: error reading '/tmp/git-private.key': Inappropriate ioctl for device\ngpg: import from '/tmp/git-private.key' failed: Inappropriate ioctl for device\ngpg: Total number processed: 0\ngpg:              unchanged: 1\ngpg:       secret keys read: 1\n\n    at ChildProcess.<anonymous> (/opt/containerbase/tools/renovate/35.143.0/node_modules/renovate/lib/util/exec/common.ts:99:11)\n    at ChildProcess.emit (node:events:525:35)\n    at ChildProcess.emit (node:domain:489:12)\n    at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)`

**Reply by rarkins** [reply]
It's a GPG error, so best to check with Google

**Reply by weikangchia** [reply]
Sure, anyway thanks for your help. I have searched online and I think GitLab bot gpg signing is not possible yet so I will fall back to using my own account.

References:
- https://gitlab.com/gitlab-org/gitlab/-/issues/387536
- https://gitlab.com/gitlab-org/gitlab/-/issues/321327

**Reply by ChristianCiach** [reply]
I just want to quickly confirm this behavior. I encountered the exact same issue when I tried to setup renovate by using a token of a Gitlab bot account. This is why I had to create a proper "user" in Gitlab, which is slightly unfortunate because normal users a billed as a "seat". But, well, it's more than worth it ;)

Using a proper user for Renovate comes with some other advantages. For example, you can use the Renovate logo as the user avatar, which (I believe) is not possible when using bot users.
